src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java

```diff
@@ -262,8 +269,11 @@ private Panel getImgViewer() {
                 return (Component) noFullImageLabel;
             }
         }).setCaption("Full Image");
+        Column<ImageInfo, Button> imageMetadataColumn = imageInfoGrid.addColumn(imageInfo -> metadataButton(imageInfo.getImageId())).setCaption("Metadata");
```

Comment:
Make the column type more general. You are using a ComponentRenderer so you can also use any vaadin component you want to. Also exception handlin has to be done here because we deal with an event based execution.
```suggestion
        Column<ImageInfo, Component> imageMetadataColumn = imageInfoGrid.addColumn(imageInfo -> {
            // Exceptions need to be handled here since they are event based and do not bubble up
            try {
                return (Component) metadataButton(imageInfo.getImageId());
            } catch (Exception e) {
                LOG.error("Could not create metadata component for imageId: " + imageInfo.getImageId());
                LOG.debug(e);
                Label noMetadataLabel = new Label("");
                return (Component) noMetadataLabel;
            }).setCaption("Metadata");
```